consensus/merge/merge.go

```diff
@@ -235,7 +234,7 @@ func (s *Merge) verifyHeader(chain consensus.ChainHeaderReader, header, parent *
 
 	skipGasLimit := false
 	if auraEngine, ok := s.eth1Engine.(*aura.AuRa); ok {
-		skipGasLimit = auraEngine.HasGasLimitContract()
+		skipGasLimit = auraEngine.HasGasLimitContract() && !misc.IsPoSHeader(header)
```

Comment:
`(s *Merge) verifyHeader` is called only for PoS headers, so `!misc.IsPoSHeader(header)` is always false.